manifests/plugin.pp

```diff
@@ -22,16 +22,23 @@
   $config_content  = undef,
   $update_url      = undef,
   $plugin_dir      = '/var/lib/jenkins/plugins',
-  $username        = 'jenkins',
-  $group           = 'jenkins',
   $enabled         = true,
   $create_user     = true,
   $source          = undef,
 ) {
   include ::jenkins::params
 
-  $plugin            = "${name}.hpi"
-  $plugin_parent_dir = inline_template('<%= @plugin_dir.split(\'/\')[0..-2].join(\'/\') %>')
+  $plugin     = "${name}.hpi"
+  $homedir    = $::jenkins::params::homedir
+  $jenkinsdir = $::jenkins::params::jenkinsdir
+  $username   = $::jenkins::params::jenkins_user
+  $group      = $::jenkins::params::jenkins_group
+
+  $real_plugin_dir = $plugin_dir ? {
+    '/var/lib/jenkins/plugins' => $::jenkins::params::default_plugin_dir,
+    default                    => $plugin_dir,
+  }
```

Comment:
The default value being assign is not the newly created variable named `default_plugin_dir` -- the variable naming is difficult to understand.  At a minimum, defined type's parameter default for $plugin_dir should be changed to `undef` and I think $default_plugin_dir should be renamed (if it is needed at all).